test/test_memory.cpp

```diff
@@ -172,20 +172,109 @@ TEST_F(MemoryTest, RefineCachePolicy) {
 }
 
 class TMATest : public NVFuserTest {
+ protected:
   void SetUp() override {
-    // requires Hopper or newer
-    if (!deviceMajorMinorCheck(9)) {
+    if (cudaArchGuardShouldSkip(9, 0)) {
       GTEST_SKIP() << "skipping tests on pre-Hopper GPUs";
     }
     NVFuserTest::SetUp();
   }
 };
 
-TEST_F(TMATest, LoadCompleteTensor1D) {
+// Check that there is an xor "^" somewhere in the kernel
+class XorFinder : private kir::IrVisitor {
```

Comment:
Didn't we have something like `getAllExprs` with a kernel IR?

Reply:
Are you referring to `ir_utils::flattenScopedExprs`? This might simplify the logic a little bit, but I don't think it will make this piece of code significantly simpler.

The complication here is, for a swizzled kernel, due to index hoisting, we might have something like:
```C++
i2 = i1 ^ i0;
T0[i3] = T0[i2];
```
or something like:
```C++
T0[i1 ^ i0] = T0[i2]
```
so we need to make sure that we are checking both the "scoped exprs" and the exprs in tensor index for xor.

Also, exprs in tensor index must be checked recursively. For example, if we have `i1 + (i2 - (i3 * i4 ^ i5))`, we will have to `handle(+)`, `handle(-)`, `handle(*)` before reaching `handle(^)`, so we need to implement this recursive algorithm anyway.

Reply:
Maybe next time we need something like this we should add a util function to grab all exprs recursively.